### PR TITLE
zynqlinux/copy_to_sd_card.sh: Fail with error message if SD card not mounted

### PR DESCRIPTION
**Pull request by andreaskurth**
Not urgent but good to have in next patch release IMO.

### CODE REVIEW

**Review by vogelpi — CHANGES_REQUESTED** [review]
It's a good idea to do this check, @accuminium . However, your suggestion just checks if the mount point exists. As CentOS creates the mount point dynamically, it will work. But I am not sure about Ubuntu.

The safe way is to check if (after finding out the mount point exists) something is actually mounted in the mount point through `/proc/mounts`

File: linux/zynqlinux/copy_to_sd_card.sh [hunk 1]
```diff
@@ -6,6 +6,12 @@ if [[ -z "${SD_BOOT_PARTITION}" ]]; then
   SD_BOOT_PARTITION="/run/media/${USER}/ZYNQ_BOOT"
 fi
```
Comment (vogelpi) [inline linux/zynqlinux/copy_to_sd_card.sh:8]: NOT_MOUNTED=`grep -qs $SD_BOOT_PARTITION /proc/mounts`

File: linux/zynqlinux/copy_to_sd_card.sh [hunk 2]
```diff
@@ -6,6 +6,12 @@ if [[ -z "${SD_BOOT_PARTITION}" ]]; then
...
 
+if [[ ! -d "${SD_BOOT_PARTITION}" ]]; then
+  echo "Error: Could not open '${SD_BOOT_PARTITION}'!"
+  echo "Please make sure your SD card is mounted and adjust 'SD_BOOT_PARTITION' to its mount point."
+  exit 1
+fi
```
Comment (vogelpi) [inline linux/zynqlinux/copy_to_sd_card.sh]: else if [ $NOT_MOUNTED -eq 1 ]; then
  echo "Error: '$SD_BOOT_PARTITION' not mounted."
  exit 1
fi

**Comment by andreaskurth** [comment]
Thanks for the suggestion, Pirmin! I changed it to `grep` `/proc/mounts`, but I am not sure if I understood your intentions regarding and control flow (the `else if`)?

I also pushed a small commit that puts the copy commands into a single one and successfully tested the two commits.

**Review by vogelpi — APPROVED** [review]

**Comment by vogelpi** [comment]
I just wanted to give you an idea. But your solution is more elegant. Thanks for implementing the changes and testing them.

**Pull request merged**